test/e2e/specs/editor/various/block-switcher-test.spec.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Block Switcher', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Block variation transforms', async ( { editor, page } ) => {
+		// This is the `stack` Group variation.
+		await editor.insertBlock( {
+			name: 'core/group',
+			attributes: {
+				layout: {
+					type: 'flex',
+					orientation: 'vertical',
+				},
+			},
+			innerBlocks: [
+				{
+					name: 'core/paragraph',
+					attributes: { content: '1' },
+				},
+			],
+		} );
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:group {"layout":{"type":"flex","orientation":"vertical"}} -->
+<div class="wp-block-group"><!-- wp:paragraph -->
+<p>1</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:group -->`
+		);
+		// Transform to `Stack` variation.
+		await editor.clickBlockToolbarButton( 'Stack' );
+		await expect(
+			page.locator( 'role=menuitem[name="Stack"i]' )
+		).toBeHidden();
+		await page.click( 'role=menuitem[name="Row"i]' );
+		expect( await editor.getEditedPostContent() ).toBe(
+			`<!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+<div class="wp-block-group"><!-- wp:paragraph -->
+<p>1</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:group -->`
+		);
```

Comment:
We don't really need to test the full output of the content. Again, similar tests should already cover this. See this [guide](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md) for more info. For the purpose of this test, we only care about whether it successfully transforms into a different variation, so checking that alone should be enough:

```suggestion
		await expect( editor.getBlocks() ).toMatchObject( [ {
			name: 'core/group',
			attributes: { flexWrap: 'nowrap', orientation: undefined /* optional */ }
		} ] );
```

Reply:
Another valid option would be to do a diff between before and after the transformation and compare that alone. The diff should only mark the changes between them, and this is exactly what we want to test here! I don't think we have a general-purpose `diff` helper for this task yet though, so you might have to be creative or just resort to the `toMatchObject` solution 😅 .